crates/sui-json/src/lib.rs

```diff
@@ -144,16 +144,29 @@ impl SuiJsonValue {
             if let Some(s) = try_parse_string(layout, bytes) {
                 json!(s)
             } else {
-                let move_value = bcs::from_bytes_seed(layout, bytes)?;
-                move_value_to_json(&move_value).unwrap_or_else(|| {
-                    // fallback to array[u8] if fail to convert to json.
-                    JsonValue::Array(
-                        bytes
-                            .iter()
-                            .map(|b| JsonValue::Number(Number::from(*b)))
-                            .collect(),
-                    )
-                })
+                let result = bcs::from_bytes_seed(layout, bytes).map_or_else(
```

Comment:
@bmwill and @dariorussi should check me, but this might hang (instead of fail) with the wrong input. We should definitely have tests for various kinds of "too big" and "too small" inputs to be sure.

Reply:
Thanks for calling this out Sam! I'm not very familiar with what would qualify as a too big or too small input but, perhaps instead of trying to map to a JsonValue::Array we can just return the bytes as a string if serde fails or times out?